Integrate rotate backups

jan-brinkmann opened the pull request:
This integrates the package [rotate-backups](https://github.com/xolox/python-rotate-backups) in docker-volume-backup.

As already proposed in the currend README.md, rotate-backups allows to preserve essential backups and eliminates redundant backups. The backup routine of docker-volume-backup now triggers rotate-backups after the local backup has been stored in the archive folder if environmental variable `ROTATE_BACKUPS: "true"`. A dry-run is enabled if `ROTATE_BACKUPS: "dry-run"`.

The default rotate-backup configuration looks like this:
```
[/archive]
daily = 7
weekly = 4
monthly = 12
yearly = always
ionice = idle
```
It can be overwritten by mounting an own configuration:
```
volumes:
  - .rotate-backups.ini:/config/.rotate-backups.ini
```
Unfortunatelly, this increases the image size from 616.5 MB to 926.6 MB (ARM64). This may primarily be as I removed the `--no-install-recommends` paramater from the `apt-get install` command in the Dockerfile. In this command, `python-pip` and some other packages must be installed in order to `pip install --no-cache-dir rotate-backups`. Maybe the image size can be reduced if all necessary packages from `apt-get` are selected manually. I am grateful for any ideas.

jareware reviewed (commented):
Hmm, this is clearly a useful feature!

I do wonder about the image size, though. It's not very slim to begin with, and this is a not-insignificant increase. 🤔 

> I am grateful for any ideas.

Since everyone will have slightly different preferences for how they'd like to manage their backups rotation (and you kindly included a way to override the config file!) I wonder if it'd make more sense to either:

1. Suggest people use another container (or a cronjob on the host) to manage rotation, or
2. Allow mounting some kind of "hook script" into the container, which would be invoked at the correct time? This would of course mean that you wouldn't have the `rotate-backups` package available in the container, but you could either:
    * Install it in the hook script if it's missing (a bit unorthodox, but would probably work just fine), or
    * Extend the image and install it yourself during `docker build`

Thoughts?

jan-brinkmann commented:
@jareware, thanks for the review!
To be honest, I was asking for ideas regarding reduction of the image size while including `rotate-backups` inse the image. :D

Here are my thoughts:

> Suggest people use another container (or a cronjob on the host) to manage rotation

If one has to start an additional container in order to use `rotate-backups`, much more drive space and main memory will be consumed in total. This might be fair as these consumption comes with an additional feature.
A cronjob on the host requires `rotate-backups` to be installed on the host. This is not always possible.
If installing `rotate-backups` on the host is possible, one could trigger `rotate-backups` on the host when when providing `SSH` access. We already have that due to the `SCP` feature. Maybe I will give it a try. :)

> Install it in the hook script if it's missing (a bit unorthodox, but would probably work just fine)

I modified the `entrypoint.sh` in such a ways that `rotate-backups` is installed if `ROTATE_BACKUPS` is set. The functionallity works as expected, but starting the container takes some time (on a Raspberry Pi 4) since round about 300MB have to be downloaded when the container is created. Now, the main drawback comes from the fact that `docker-compose down` removes all associated containers. Thus, the 300MB have to be downlaoded everytime one runs `docker-compose down` and `docker-compose up`. Imho, this results in a very bad user experience.

> Extend the image and install it yourself during `docker build`

This is what I am doing since I implemented the feature. I love it, hence, I like to share it. :)

jareware commented:
@jan-brinkmann do you think this is still needed, as #43 allows solving backup rotation pretty elegantly IMO outside of this image altogether?
